### ISSUE TITLE
Same form for creating new posts & editing

### ISSUE THREAD

**Issue by JiveDig** [Desertsnowman/Caldera-Forms#192]
I'd like to only manage one form for both creating and editing a post. Not sure how yet, but some sort of condition for the Post ID field to know when to edit an ID and when to create a new one. Or maybe another method to simplify things.

This might even be possible now already, it's just not obvious how, and I can't visualize how to accomplish this other than duplicating the form.

<img width="1092" alt="conditional-magic-tags" src="https://cloud.githubusercontent.com/assets/540990/9289891/2d7673e0-434c-11e5-9d78-3a62ac782072.png">

**Comment by Shelob9** [comment]
@JiveDig A workaround, that is suboptimal, would be two instances of this processor. One with post ID, set and one without, then make them conditional based on a "create new/ edit" checkbox.

**Comment by JiveDig** [comment]
@Shelob9 That could work in some instances. I was considering building a small add-on/plugin to do something like this tu.be/4ehOO9tJ_UY?t=56s - I did that with ACF Pro, but it's a bit code-heavy for my liking. So it would be a separate form for editing then it would be for posting. But in admin i'd like to only maintain 1 form. Not a deal breaker, but would be a nice option.

**Comment by JiveDig** [comment]
Just thinking about this more. Wondering if `post_id` as a shortcode parameter would work here.
Something like this for posting:
```
[caldera_form id="CF55cd33c782100" post_id="new"]
```

And something like this for editing:
```
[caldera_form id="CF55cd33c782100" post_id="embed_post_id"]
```

Another potential addition to this...
We could ultimately use (new?) conditions for hiding/showing specific fields if post_id is or is not 'new' (or similar). For times when you don't want to allow a user to edit the post title or category after it's been posted, as an example.

All this could be possible using the same form.

**Comment by Desertsnowman** [comment]
@JiveDig with the form be used on separate pages?

**Comment by JiveDig** [comment]
@Desertsnowman Yes... new post would have it's own page (/new-post/). Editing posts would be a slide out thing like in the youtube example (or however people wanna do it). It would be inserted into the single-{post-type}.php or whatever template.

**Comment by Desertsnowman** [comment]
Can you give a user case example?

**Comment by Desertsnowman** [comment]
Correction use case example

**Comment by JiveDig** [comment]
I build a site to allow registered users to submit posts from the front end, and edit them from the front end. User has no access to toolbar or dashboard.

User logs in via a front end login form (Caldera Users add-on FTW).

User goes to mysite.com/new-post/ where `[caldera_form id="CF55cd33c782100" post_id="new"]` allows them to submit a new post.

After successful submission, User is redirected to their new post automatically.

User notices a spelling mistake in the content. Because they are logged in and the author of the post, my custom code shows an 'Edit' button next to the post title.

User clicks 'Edit' and a slide out window (or modal, or something I build), shows with `[caldera_form id="CF55cd33c782100" post_id="embed_post_id"]` in it, allowing User to editing their post.

**Comment by JiveDig** [comment]
Another example might be to show User a list of their post titles on a protected page. They click 'Edit' next to one of the titles and it brings them to a static page `mysite.com/edit-post/?post_id=1234`

The form could grab the query arg (post_id) and use that as the post_id for which to edit.

**Comment by Desertsnowman** [comment]
so the editing form will be on the same post they are editing?

**Comment by JiveDig** [comment]
In this example, yes. That's the best way I could come up with to do this. But i'm open to other ideas.

**Comment by JiveDig** [comment]
This is all currently possible if I duplicate my form and set post id of the form to embed_post:ID.  My hope was to maintain all of this from 1 form, since the form could potentially be complex.

**Comment by Desertsnowman** [comment]
Add a variable. Call it posttype. Set the value to {embed_post:post_type}
Add a second 'save as post type' processor and copy the firsts settings except set the post Id to {embed_post:ID} .

Set the first processor conditions to "use if variable:posttype  = page ( or the post type the creating instance of the form is on)

On the second processor,  set the conditions to use if variable:posttype = the type you're editing.

That should do it.

**Comment by JiveDig** [comment]
Brilliant. I was just thinking to do this with author. If embed_post:author == user_id...  but those aren't existing variables, and i'm not sure they even should be at this point.

The post type method is a great idea, and the only time it wouldn't work is if anyone would allow page creation/editing. If that were the case, I could possibly use the actual post IDs instead of post type. If embed_post:ID = (some fixed ID), then do my thing.

Great head start... thanks for thinking this out.

**Comment by JiveDig** [comment]
Oh wait... does {embed_post:} give you all the data? I need to test. I think going the author route might be safest if that data is already accessible.

**Comment by Desertsnowman** [comment]
You can access most fast in {embed_post:*} even custom fields. I.e {embed_post:my_field} 

Keep that in mind.

**Comment by JiveDig** [comment]
Okay, hopefully i'm not being a burden here...

I've tested your suggested method with post_type and also with post ID.

Any time I have 2 processors, even with the different Use or Don't Use settings, the form is always showing the embed page content. The only way to stop it is to remove 1 of the processors. But then I can't post new or edit existing.

**Comment by Desertsnowman** [comment]
Bother!! I'm going to have to test it. I'm going to bed now. I'll test in the morning and hopefully give you a better solution.

**Comment by JiveDig** [comment]
Thanks @Desertsnowman. After playing with variables and your head start, I think this is the best method. It just seems like the Use/Don't Use is not respecting my settings or variables.

**Comment by d10n-de** [comment]
Hi folks, this one is still open, right? I'd love to find a solution for the exact same task: one single form for editing and creating a (custom-)post(-type). Any tips, tricks, ideas - apart from what was discussed here?

Thanks in advance and fourty thumbs up for this awesome piece of software!! 

Cheers, Dirk 
from Hamburg/Germany
